app/src/main/java/woowacourse/movie/alarm/MovieBroadcastReceiver.kt

```diff
@@ -0,0 +1,77 @@
+import android.app.PendingIntent
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.app.NotificationCompat
+import woowacourse.movie.database.TicketDatabase
+import woowacourse.movie.list.view.SettingFragment.Companion.NOTIFICATION_KEY
+import woowacourse.movie.list.view.SettingFragment.Companion.SHARED_PREFERENCE_SETTING
+import woowacourse.movie.ticket.model.DbTicket
+import woowacourse.movie.ticket.view.MovieTicketActivity
+import java.util.concurrent.CountDownLatch
+
+class MovieBroadcastReceiver : BroadcastReceiver() {
+    private lateinit var tickets: List<DbTicket>
+    private var movieTitle: String? = null
+    override fun onReceive(context: Context, intent: Intent?) {
+        val sharedPreference =
+            context.getSharedPreferences(SHARED_PREFERENCE_SETTING, AppCompatActivity.MODE_PRIVATE)
+        val isNotificationGranted = sharedPreference.getBoolean(NOTIFICATION_KEY, false)
+        movieTitle = intent?.getStringExtra(MovieTicketActivity.EXTRA_MOVIE_TITLE_KEY)
+        val latch = CountDownLatch(1)
+        loadTicketsFromDb(context, latch)
+        latch.await()
+        if (!isNotificationGranted) return
+        makeNotification(context)
+    }
+
+    private fun loadTicketsFromDb(
+        context: Context,
+        latch: CountDownLatch
+    ) {
+        Thread {
+            val ticketDb = TicketDatabase.getDatabase(context)
+            tickets = ticketDb.ticketDao().getAll()
+            latch.countDown()
+        }.start()
+    }
+
+    private fun makeNotification(context: Context) {
+        val manager = context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
+        val builder: NotificationCompat.Builder
+        val channelId = "one-channel"
+        val channelName = "My Channel One"
+        val channel =
+            NotificationChannel(channelId, channelName, NotificationManager.IMPORTANCE_DEFAULT)
+        manager.createNotificationChannel(channel)
+        builder = NotificationCompat.Builder(context, channelId)
+        val pendingIntent = makePendingIntent(context)
+        setBuilder(builder, pendingIntent)
+        manager.notify(1, builder.build())
+    }
+
+    private fun makePendingIntent(context: Context): PendingIntent? {
+        val ticketIntent = MovieTicketActivity.newTicketActivityInstance(context, tickets, 1)
+        return PendingIntent.getActivity(context, 10, ticketIntent, PendingIntent.FLAG_MUTABLE)
+    }
+
+    private fun setBuilder(
+        builder: NotificationCompat.Builder,
+        pendingIntent: PendingIntent?,
+    ) {
+        builder.setSmallIcon(android.R.drawable.ic_popup_reminder)
+        builder.setContentTitle(MESSAGE_RESERVATION_NOTIFICATION)
+        builder.setContentText(MESSAGE_SCREENING_INFORMATION.format(movieTitle))
+        builder.setContentIntent(pendingIntent)
+    }
```

Comment:
```suggestion
    private fun buildNotification(
        context: Context,
        pendingIntent: PendingIntent,
    ): Notification {
        return NotificationCompat.Builder(context, CHANNEL_ID)
            .setSmallIcon(android.R.drawable.ic_popup_reminder)
            .setContentTitle(MESSAGE_RESERVATION_NOTIFICATION)
            .setContentText(MESSAGE_SCREENING_INFORMATION.format(movieTitle))
            .setContentIntent(pendingIntent)
            .build()
    }
```

보통 빌드는 하나의 함수 내에서 이루어지게 됩니다.
빌드에 필요한 요소를 파라미터로 넘기고 내부적으로 모든 빌드 작업을 한 뒤에 결과물을 반환하도록 하는건 어떨까요?

Reply:
제안주셔서 감사합니다! 확실히 더 깔끔하네요! 😘